lib/local-links-manager/import/local_authorities_importer.rb

```diff
@@ -38,6 +38,7 @@ def authorities_from_mapit
             Rails.logger.info("Updating authority #{mapit_la[:name]} (gss #{mapit_la[:gss]})")
             la.update_attributes!(
               name: mapit_la[:name],
+              snac: mapit_la[:snac],
```

Comment:
Now that we're updating the snac too, we're setting the same attributes on update and create. We could use `first_or_initialize.tap` to unify these cases so that the same code is used for both which will help us avoid accidentally handling them differently in future:

```ruby
LocalAuthority.where(gss: mapit_la[:gss]).first_or_initialize.tap do |la|
  verb = la.persisted? ? "Updating" : "Creating"
  Rails.logger.info("#{verb} authority #{mapit_la[:name]} (gss #{mapit_la[:gss]})")

  la.name = mapit_la[:name]
  la.snac = mapit_la[:snac]
  la.tier = mapit_la[:tier]
  la.save!
end
```

The `tap` is there so that the block is run in both cases - `first_or_initialize` only uses the block when initializing and not when it finds an existing record.

Transition [uses this approach](https://github.com/alphagov/transition/blob/master/lib/transition/import/site_yaml_file.rb#L72-L90) in a very similar context - I know not everyone is fond of `tap`, though, so other opinions are available...